src/app/services/storage.service.ts

```diff
@@ -240,4 +240,16 @@ export class StorageService {
 
     return path.indexOf('/') < 0;
   }
+
+  // Takes a string with fuzzy units like 20Mib, 20k, 20Gb, returns number in bytes
+  convertHumanStringtoNum(str: string) {
+    if(!str) { return 0 }; // if empty, return the default 0, which is unlimited
+    const letters = str.replace( /[^a-zA-Z]/g, '');
+    if (letters[0] && !letters[0].toLowerCase().match(/b|k|m|g|t/)) { return NaN };
+    const powersOf1024 = {b: 1, k: 1024, m: 1024**2, g: 1024**3, t: 1024**4};
+    const unit = str.toLowerCase().match(/[b|k|m|g|t]/) || 'b';
+    let num;
+    const value = (num = str.match(/^\s*(\d+)/)) ? num[1] : NaN;
+    return parseInt(value) * powersOf1024[unit.toString()];
```

Comment:
Not sure why the toString call is here.  That list should return a number.

Reply:
This is a Typescript thing. TS says that unit is of type RegExpMatchArray. I can't use parseInt() to make it an integer either, but I can change it to a string, and JS knows what to do with a string here.

![image](https://user-images.githubusercontent.com/9504493/62564858-a31e9e80-b853-11e9-97ad-078f0802d286.png)